compiler-plugin/src/test/kotlin/arrow/meta/quotes/transform/plugins/TransformNewSourcePlugin.kt

```diff
@@ -0,0 +1,98 @@
+package arrow.meta.quotes.transform.plugins
+
+import arrow.meta.Meta
+import arrow.meta.Plugin
+import arrow.meta.invoke
+import arrow.meta.phases.CompilerContext
+import arrow.meta.quotes.Transform
+import arrow.meta.quotes.classDeclaration
+import arrow.meta.quotes.classorobject.ClassDeclaration
+import arrow.meta.quotes.plus
+import org.jetbrains.kotlin.psi.KtClass
+
+val Meta.transformNewSource: List<Plugin>
+  get() = listOf(transformNewSourceSingleGeneration, transformNewSourceWithManyTransformation, transformNewSourceMultipleGeneration)
+
+private val Meta.transformNewSourceSingleGeneration: Plugin
+  get() = "Transform New Source" {
+    meta(
+      classDeclaration({ name == "NewSource" }) {
+        Transform.newSources(
+          """
+            package arrow
+            //metadebug
```

Comment:
In this case, I think it wouldn't be necessary to add the debug command for Arrow Meta in the plugin to get the generated sources again. I would try to compare the expected generated code with the actual generated code.